packages/dataviews/src/filter-and-sort-data-view.ts

```diff
@@ -140,10 +140,9 @@ export function filterSortAndPaginate< Item >(
 		} );
 		if ( fieldToSort ) {
 			filteredData.sort( ( a, b ) => {
-				const valueA = fieldToSort.getValue( { item: a } ) ?? '';
-				const valueB = fieldToSort.getValue( { item: b } ) ?? '';
-
-				if ( fieldToSort.type === 'integer' ) {
+				if (
+					[ 'integer', 'text' ].includes( fieldToSort.type ?? '' )
+				) {
```

Comment:
I'm not sure we need this. We can just move the "fallback" sorting to the sorting function you have as a fallback in `getFieldTypeDefinition`

Reply:
Oh, nice. That's actually better https://github.com/WordPress/gutenberg/pull/64168/commits/2d80886dbdfc5560c0528bc39648db299bd5b20e